tests/integration/test_mean.py

```diff
@@ -0,0 +1,166 @@
+    (1,),
+    (DIM,),
+    (1, 1),
+    (1, DIM),
+    (DIM, 1),
+    (DIM, DIM),
+    (1, 1, 1),
+    (DIM, 1, 1),
+    (1, DIM, 1),
+    (1, 1, DIM),
+    (DIM, DIM, DIM),
+)
+
+
+def gen_out_shape(size, axis):
+    if axis is None:
+        return ()
+    if axis < 0:
+        axis += len(size)
+    if axis >= 0 and axis < len(size):
+        return size[:axis] + size[axis + 1 :]
+    else:
+        return -1
+
+
+@pytest.mark.parametrize("arr", ([], [[], []]))
+def test_empty_arr(arr):
+    res_np = np.mean(arr)
+    res_num = num.mean(arr)
+    assert np.isnan(res_np) and np.isnan(res_num)
+
+
+def test_scalar():
+    res_np = np.mean(10)
+    res_num = num.mean(10)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("size", NO_EMPTY_SIZE)
+def test_basic(size):
+    arr_np = np.random.randint(-5, 5, size=size)
+    arr_num = num.array(arr_np)
+    res_np = np.mean(arr_np)
+    res_num = num.mean(arr_num)
+    np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("axis", ((-3, -1), (-1, 0), (-2, 2), (0, 2)))
+def test_axis_tuple(axis):
+    # In Numpy, it pass
+    # In cuNumeric, it raises NotImplementedError
+    size = (3, 4, 7)
+    arr_np = np.random.randint(-5, 5, size=size)
+    arr_num = num.array(arr_np)
+    out_np = np.mean(arr_np, axis=axis)
+    out_num = num.mean(arr_num, axis=axis)
+    assert np.array_equal(out_np, out_num)
+
+
+@pytest.mark.parametrize(
+    "keepdims", (False, pytest.param(True, marks=pytest.mark.xfail))
+)
+@pytest.mark.parametrize("size", NO_EMPTY_SIZE)
+def test_axis_keepdims(size, keepdims):
+    # In all cases except for (1,) ,(DIM,), (DIM,DIM) and (DIM, DIM, DIM),
+    # in Numpy, pass
+    # in cuNumeric, raises AssertionError
+    arr_np = np.random.randint(-5, 5, size=size)
+    arr_num = num.array(arr_np)
+    ndim = arr_np.ndim
+    for axis in range(-ndim, ndim):
+        out_np = np.mean(arr_np, axis=axis, keepdims=keepdims)
+        out_num = num.mean(arr_num, axis=axis, keepdims=keepdims)
+        assert np.array_equal(out_np, out_num)
+
+
+@pytest.mark.parametrize("array_dt", (np.int32, np.float32, np.complex64))
+@pytest.mark.parametrize("dt", (np.int32, np.float32, np.complex64))
+@pytest.mark.parametrize("size", NO_EMPTY_SIZE)
+def test_dtype(size, array_dt, dt):
+    arr_np = np.random.randint(-5, 5, size=size).astype(array_dt)
+    arr_num = num.array(arr_np)
+    res_np = np.mean(arr_np, dtype=dt)
+    res_num = num.mean(arr_num, dtype=np.dtype(dt))
+    np.array_equal(res_np, res_num)
+
+
+@pytest.mark.parametrize("out_dt", (np.int32, np.float32))
+@pytest.mark.parametrize("size", NO_EMPTY_SIZE)
+def test_out(size, out_dt):
+    arr_np = np.random.randint(-5, 5, size=size)
+    arr_num = num.array(arr_np)
+    ndim = arr_np.ndim
+    for axis in (-1, ndim - 1, None):
+        out_shape = gen_out_shape(size, axis)
+        out_np = np.zeros(out_shape, dtype=out_dt)
+        out_num = num.ones(out_shape, dtype=out_dt)
```

Comment:
why `zeros` for one and `ones` for the other? `empty` seems like it should be sufficient?

Reply:
Fixed. Thanks.